### PR TITLE
Refactor Modules for Inbound Authentication

### LINKED ISSUE #15385: Update Module Structure for Ballerina Inbound Authentication
**Description:**
Need to update the module structure for ballerina inbound authentication as discussed in [1]

[1] https://groups.google.com/d/msg/ballerina-dev/7SYDiKeF8p8/3GNncS9dAwAJ

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
The current `ballerina/auth` module has implementation related to Config Auth Store, JWT, LDAP etc. So, with this PR, `ballerina/jwt` and `ballerina/ldap` modules are newly created. Please refer to the discussion @ballerina-dev [1].

Also, the HTTP authn handlers were renamed in a generic manner as follows:

| Old syntax | New syntax |
| --- | --- |
| `http:JwtAuthnHandler` | `http:BearerAuthHeaderAuthnHandler` |
| `http:BasicAuthnHandler` | `http:BasicAuthHeaderAuthnHandler` |

[1] https://groups.google.com/d/msg/ballerina-dev/7SYDiKeF8p8/3GNncS9dAwAJ

Fixes #15385

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [x] Checked Tooling Support (#15436)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/jwt/src/main/ballerina/jwt/utils.bal [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Log, prepare and return the `error`.
+#
+# + message - Error message
+# + err - `error` instance
+# + return - Prepared `error` instance
+function prepareError(string message, error? err = ()) returns error {
+    log:printError(message, err = err);
+    error preparedError = error(JWT_ERROR_CODE, { message: message, reason: err.reason() });
+    return preparedError;
+}
```
Comment (Bhashinee) [inline stdlib/jwt/src/main/ballerina/jwt/utils.bal]: Shall we add a new line?

File: stdlib/ldap/assembly/source.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<!--
+  ~ /*
+  ~ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Bhashinee) [inline stdlib/ldap/assembly/source.xml]:
```suggestion
  ~ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: stdlib/auth/src/main/ballerina/auth/Module.md [hunk 3]
```diff
@@ -1,35 +1,18 @@
...
 
 An authentication provider defines an authentication scheme that could be used to authenticate endpoints. The `auth:AuthStoreProvider` acts as the interface for all the authentication providers. Any type of implementation such as LDAP, JDBC, and file-based should be object-equivalent.
 
-By default, there are three implementations of the `auth:AuthProvider`. They are:
-1. The `auth:ConfigAuthStoreProvider`, which authenticates based on usernames and passwords stored in a configuration file.
-2. The `auth:JWTAuthProvider`, which authenticates by validating a JWT.
-3. The `auth:LdapAuthStoreProvider`, which authenticates based on the user credentials stored in an active directory or an LDAP.
-
-When creating a new authentication provider, there is a function that needs to be implemented. 
+When creating a new authentication provider, there is a function that needs to be implemented.
```
Comment (praneesha) [inline stdlib/auth/src/main/ballerina/auth/Module.md]:
```suggestion
When creating a new authentication provider, you need to implement the below function.
```

File: stdlib/auth/src/main/ballerina/auth/Module.md [hunk 4]
```diff
@@ -1,35 +1,18 @@
...
 - `authenticate` : Authenticates the user based on a credential, which can be username/password, or a token such as JWT.
```
Comment (praneesha) [inline stdlib/auth/src/main/ballerina/auth/Module.md]:
```suggestion
- `authenticate` : Authenticates the user based on the user credentials (i.e., the username/password) or a token such as JWT.
```

File: stdlib/auth/src/main/ballerina/auth/Module.md [hunk 5]
```diff
@@ -1,35 +1,18 @@
...
 
-### Config Auth Store Provider
-
-The `auth:ConfigAuthStoreProvider` is an implementation of the `auth:AuthProvider` interface, which uses the Ballerina configuration file to store usernames, passwords, scopes, and relevant associations.
-
-A user is denoted by a section in the configuration file. The password and the scopes assigned to the user are denoted as keys under the relevant user section as shown below. 
+By default, there is `auth:ConfigAuthStoreProvider`, which authenticates based on usernames and passwords stored in a configuration file. The `auth:ConfigAuthStoreProvider` is an implementation of the `auth:AuthProvider` interface, which uses the Ballerina configuration file to store usernames, passwords, scopes, and relevant associations. A user is denoted by a section in the configuration file. The password and the scopes assigned to the user are denoted as keys under the relevant user section as shown below.
```
Comment (praneesha) [inline stdlib/auth/src/main/ballerina/auth/Module.md]:
```suggestion
By default, the`auth:ConfigAuthStoreProvider` authenticates based on usernames and passwords stored in a configuration file. The `auth:ConfigAuthStoreProvider` is an implementation of the `auth:AuthProvider` interface, which uses the Ballerina configuration file to store usernames, passwords, scopes, and relevant associations.

A user is denoted by a section in the configuration file. The password and the scopes assigned to the user are denoted as keys under the relevant user section as shown below.
```

File: stdlib/jwt/src/main/ballerina/jwt/Module.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
+## Module Overview
+
+This module provides JWT authentication provider, which can be used to authenticate with a JWT token; and functionality related to issuing a JWT and validating JWT.
```
Comment (praneesha) [inline stdlib/jwt/src/main/ballerina/jwt/Module.md]:
```suggestion
This module provides a JWT authentication provider, which can be used to authenticate using a JWT token and the functionality related to issuing a JWT token and validating it.
```

File: stdlib/ldap/src/main/ballerina/ldap/Module.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+## Module Overview
+
+This module provides LDAP authentication provider, which can be used to authenticate with LDAP credentials.
```
Comment (praneesha) [inline stdlib/ldap/src/main/ballerina/ldap/Module.md]:
```suggestion
This module provides an LDAP authentication provider, which can be used to authenticate using LDAP credentials.
```

**Pull request merged**